scripts/check_total_coverage.sh

```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env bash
+
+coverage_dir=$(PWD)/coverage
+combined_coverage_file=$coverage_dir/combined.txt
+COVERAGE_THRESHOLD=15.0 # percentage threshold of code coverage required
+
+echo "Generating coverage report in text format..."
+included_packages=$(go list ./... | grep -v /tests/ | grep -v '/sdk/') # not including 'tests' and 'sdk'
+go tool covdata merge -i=$coverage_dir/e2e,$coverage_dir/ut -o=$coverage_dir -pkg=${included_packages//$'\n'/,}
+go tool covdata textfmt -i=./coverage -o $combined_coverage_file
+go tool cover -func $combined_coverage_file
+
+echo "Checking total coverage..."
+
+# print current test coverage from report
+current_test_coverage=$(go tool cover -func=$combined_coverage_file  | grep -e "total" | awk '{print $3}')
```

Comment:
lets call go tool cover once (previous exec to go into a temp file)

Reply:
addressed